advisor-ui/src/pages/Signup/Signup.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState, useContext } from "react";
+import InputForm from "../../components/InputForm/InputForm";
+import axios from "axios";
+import { useNavigate, Link } from "react-router-dom";
+import { UserContext } from "../../UserContext.js";
+
+export default function Signup() {
+  const [firstname, setFirstname] = useState("");
+  const [lastname, setLastname] = useState("");
+  const [email, setEmail] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [year, setYear] = useState("");
+  const [city, setCity] = useState("");
+  const [locationState, setLocationState] = useState("");
+  const [address, setAddress] = useState("");
+
+  const navigate = useNavigate();
+  const { updateUser } = useContext(UserContext);
+
+  const handleFirstNameChange = (e) => {
+    setFirstname(e.target.value);
+  };
+
+  const handleLastNameChange = (e) => {
+    setLastname(e.target.value);
+  };
+
+  const handleUserNameChange = (e) => {
+    setUsername(e.target.value);
+  };
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  };
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+  };
+  const handleYearChange = (e) => {
+    setYear(e.target.value);
+  };
+  const handleCityChange = (e) => {
+    setCity(e.target.value);
+  };
+  const handleStateChange = (e) => {
+    setLocationState(e.target.value);
+  };
+  const handleAddressChange = (e) => {
+    setAddress(e.target.value);
+  };
+
+  const handleCreateAccount = async () => {
+    if (
+      firstname === "" ||
+      lastname === "" ||
+      username === "" ||
+      email === "" ||
+      password === "" ||
+      year === "" ||
+      city === "" ||
+      locationState === "" ||
+      address === ""
+    ) {
+      alert("Please enter all fields");
+      return;
+    }
+    const body = {
+      firstname,
+      lastname,
+      username,
+      password,
+      email,
+      year: parseInt(year),
+      city,
+      state: locationState,
+      address: address,
+    };
+    try {
+      const res = await axios.post(
+        "http://localhost:5000/api/student/create",
```

Comment:
You probably should use relative path here instead, or store these in a separate consts file - otherwise you'll need changes to deploy this.